### ISSUE TITLE
jsconfig.json is tracked as a manifest file but never read for path alias resolution

### ISSUE THREAD

**Issue by MILLERMARRU** [biomejs/biome#11282]
### Environment information

Source-level finding, not tied to a specific local environment, this is about the resolver's file-discovery logic itself, not a runtime-dependent repro. Confirmed against the current `main` branch source (not a released version-specific regression).

### What happened?

`jsconfig.json` is recognized by Biome as a manifest-priority file, but its content is never actually read for path/alias resolution, only `tsconfig.json` is.

Concretely:

1. `crates/biome_fs/src/path.rs` lists `jsconfig.json` at the same priority tier as `package.json`/`tsconfig.json`/`turbo.json`/`pnpm-workspace.yaml`:

```rust
// crates/biome_fs/src/path.rs
matches!(
    file_name,
    "package.json"
        | "tsconfig.json"
        | "jsconfig.json"
        | "turbo.json"
        | "turbo.jsonc"
        | "pnpm-workspace.yaml"
)
```

2. The module resolver that actually parses `compilerOptions.paths`/`baseUrl` (`crates/biome_package/src/node_js_package/tsconfig_json.rs`, `TsConfigJson::parse`) is only ever invoked against a file discovered by the literal name `tsconfig.json`.

3. A full-repo search for the string `"jsconfig.json"` turns up exactly three matches: the priority list above, a JSON-schema registration in `crates/biome_languages/src/json.rs` (so Biome treats it as a schema-validatable JSON file, same reason `tsconfig.json` gets schema support), and a `CHANGELOG_v1.md` mention. None of them feed into module resolution.

Practical repro: a plain JavaScript project (e.g. `create-next-app` with the JS option) using `jsconfig.json` for a `@/*` path alias, no `tsconfig.json` present. Any Biome feature that depends on resolving that import (e.g. `lint/correctness/noUnresolvedImports`) treats the aliased import as unresolved, even though the alias is valid per `jsconfig.json` and correctly resolved by the project's actual bundler (webpack/Turbopack, which do read `jsconfig.json`).

Originally surfaced in #11279, where a user also found that adding a `tsconfig.json` alongside a JS project isn't a safe workaround under Next.js specifically, since Next.js actively rewrites `tsconfig.json` on disk and auto-installs `typescript`/`@types/*` the moment the file exists, converting the project's shape rather than just adding an unused file.

### Expected result

Either:
- `jsconfig.json`'s `compilerOptions.paths`/`baseUrl` are read the same way `tsconfig.json`'s are, since `jsconfig.json` is already being tracked as a manifest file for other purposes, or
- if this is intentionally out of scope, the docs/rule descriptions for anything depending on alias resolution (e.g. `noUnresolvedImports`) should say so explicitly, since today the file being recognized-but-unread is easy to miss and reads as a bug rather than a documented limitation.
